pkg/api/graph/graph.go

```diff
@@ -114,6 +111,23 @@ type Interface interface {
 	Edges() []graph.Edge
 }
 
+type Namer interface {
+	ResourceName(obj interface{}) string
+}
+
+type namer struct{}
+
+var DefaultNamer Namer = namer{}
+
+func (namer) ResourceName(obj interface{}) string {
+	switch t := obj.(type) {
+	case uniqueNamer:
+		return t.UniqueName()
```

Comment:
DefaultNamer is only there for test cases.  There is a separate namer for
project status, which knows about resource shortcuts.

On Fri, Dec 18, 2015 at 8:21 AM, David Eads <notifications@github.com>
wrote:

> In pkg/api/graph/graph.go
> <https://github.com/openshift/origin/pull/6389#discussion_r48023860>:
>
> > @@ -114,6 +111,23 @@ type Interface interface {
> >  	Edges() []graph.Edge
> >  }
> >
> > +type Namer interface {
> > +	ResourceName(obj interface{}) string
> > +}
> > +
> > +type namer struct{}
> > +
> > +var DefaultNamer Namer = namer{}
> > +
> > +func (namer) ResourceName(obj interface{}) string {
> > +	switch t := obj.(type) {
> > +	case uniqueNamer:
> > +		return t.UniqueName()
>
> The previous intent was "resourceTypeShortName/resourceName". UniqueName
> has different constraints. As I recall, I have it doing pipes and
> namespaces.
>
> I really want to be able to copy/paste the output to kubectl and I think
> that this change makes it impossible to do that in the common case.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/6389/files#r48023860>.
>